src/map.rs

```diff
@@ -168,6 +159,19 @@ impl<'a, T> PutResult<'a, T> {
     }
 }
 
+/// The error type for the [`HashMap::try_insert`] method.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct TryInsertError<'a, V> {
+    /// A reference to the old value mapped to the key.
+    pub old: &'a V,
+    /// A reference to the new value, the one passed as an argument to
+    /// [`HashMap::try_insert`].
+    pub not_inserted: &'a V,
+}
+
+/// The return type of [`HashMap::try_insert`].
+pub type TryInsertResult<'a, V> = Result<&'a V, TryInsertError<'a, V>>;
```

Comment:
I don't think we should have this type alias.

Reply:
I created it out of frustration (of typing `Result<&'a V, TryInsertError<'a, V>>`). What about making it private?

Reply:
Hmm, that type should only appear in two places though, right? Once in `map_ref.rs` and once in `map.rs`? I don't think adding an extra type to the API is worth the single duplication. We also sadly can't make it private as it would then still appear in the function signatures, but the user wouldn't know what type it actually is.

Reply:
Oh, I see. I thought private type aliases were simply ignored by `docs.rs`.

Makes sense, we should expand our API surface unnecessarily. I'll fix this as soon as I can.

Reply:
Ah, no, sadly not. Type aliases are treated as identical to the original type, but they are not "removed". Whenever they appear, they appear as the alias, not the target type. In fact, the code probably wouldn't even compile since you'd have a private type in a public interface!